cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -185,7 +186,14 @@ public Cluster createCluster(ProvisionerContext context) throws Exception {
       }
 
       LOG.info("Creating Dataproc cluster {} with system labels {}", clusterName, systemLabels);
-      client.createCluster(clusterName, imageVersion, systemLabels);
+      ClusterOperationMetadata createOperationMeta = client.createCluster(clusterName, imageVersion, systemLabels);
+      int numWarnings = createOperationMeta.getWarningsCount();
+      if (numWarnings > 0) {
+        LOG.warn("Dataproc cluster create operation has {} warning{}", numWarnings, numWarnings > 1 ? "s" : "");
+        for (int i = 0; i < numWarnings; i++) {
+          LOG.warn("{}", createOperationMeta.getWarnings(i));
```

Comment:
If other threads log messages while this thread is iterating over the warnings, context is lost.
Should we log the warnings in batch?

Reply:
true, will make it one message.